include/multipass/ssh/sftp_client.h

```diff
@@ -18,31 +18,54 @@
 #ifndef MULTIPASS_SFTP_CLIENT_H
 #define MULTIPASS_SFTP_CLIENT_H
 
-#include <multipass/ssh/ssh_session.h>
+#include "ssh_session.h"
 
 #include <libssh/sftp.h>
 
+#include <QFlags>
+#include <filesystem>
+#include <functional>
 #include <iostream>
-#include <memory>
-#include <string>
+
+namespace fs = std::filesystem;
 
 namespace multipass
 {
 using SSHSessionUPtr = std::unique_ptr<SSHSession>;
-using SFTPSessionUPtr = std::unique_ptr<sftp_session_struct, void (*)(sftp_session)>;
+using SFTPSessionUPtr = std::unique_ptr<sftp_session_struct, std::function<void(sftp_session)>>;
+
+enum TransferFlags
+{
+    Recursive = 1,
+};
```

Comment:
A single member `enum` seems a bit strange.  Any plans for other transfer types or perhaps this is just forward thinking for the future?

Reply:
yes, there will also be a `-p` flag PR after this one :smile:

Reply:
Ok, cool, makes sense now. :slightly_smiling_face: